### PR TITLE
Refine image classification doc

### LINKED ISSUE #138: image_classificatio
In https://github.com/PaddlePaddle/book/pull/97, @gangliao updated the new demo program using V2 API. We need to update the README.en.md accordingly.

### PR DESCRIPTION
**Pull request by gangliao**
Fix #138

### CODE REVIEW

File: image_classification/README.en.md [hunk 1]
```diff
@@ -141,157 +141,87 @@ Figure 10. ResNet model for ImageNet
...
+Since ImageNet is too large to be downloaded and trained efficiently, we use CIFAR-10 (https://www.cs.toronto.edu/~kriz/cifar.html) in this tutorial. The CIFAR-10 dataset consists of 60000 32x32 color images in 10 classes, with 6000 images per class. There are 50000 training images and 10000 test images. Figure 11 shows all the classes in CIFAR-10 as well as 10 images randomly sampled from each category.
 
 <p align="center">
 <img src="image/cifar.png" width="350"><br/>
 Figure 11. CIFAR10 dataset[21]
 </p>
 
-The following command is used for downloading data and calculating the mean image used for data preprocessing.
+ `paddle.datasets` package encapsulates multiple public datasets, including `cifar`, `imdb`, `mnist`, `moivelens` and `wmt14`, etc. There's no need for us to manually donwload and preprocess CIFAR-10.
```
Comment (helinwang) [inline image_classification/README.en.md]: donwload -> download

File: image_classification/README.en.md [hunk 2]
```diff
@@ -141,157 +141,87 @@ Figure 10. ResNet model for ImageNet
...
-                    im = im.reshape(3, 32, 32)
-                    im = im[:,:,::-1]
-                    im = im.flatten()
-                im = im - settings.mean
-                yield {
-                    'image': im.astype('float32'),
-                    'label': int(lab)
-                }
-```
+After issuing a command `python train.py`, trainning is starting immediately! The details will be unpacked by the following sessions to see how it works.
```
Comment (helinwang) [inline image_classification/README.en.md]: `trainning is starting immediately!` -> `training will start immediately.`

File: image_classification/README.en.md [hunk 3]
```diff
@@ -141,157 +141,87 @@ Figure 10. ResNet model for ImageNet
...
+As alluded to in section `Overview Models`, here we provide the implementations of both VGG and ResNet models.
 
-### Model Architecture
-
-Here we provide the cofig files for VGG and ResNet models.
-
-#### VGG
+### VGG
 
 First we define VGG network. Since the image size and amount of CIFAR10 are relatively small comparing to ImageNet, we uses a small version of VGG network for CIFAR10. Convolution groups incorporate BN and dropout operations.
```
Comment (helinwang) [inline image_classification/README.en.md]: `First` -> `First,`

File: image_classification/README.en.md [hunk 4]
```diff
@@ -395,106 +313,131 @@ The following are the components of `resnet_cifar10`:
...
+```python
+trainer.train(
+    reader=reader,
+    num_passes=200,
+    event_handler=event_handler,
+    feeding=feeding)
 ```
 
-### Feature Extraction
+Here is an example log after training for one pass. The average error rates are 0.6875 on training set and 0.8852 on validation set.
```
Comment (helinwang) [inline image_classification/README.en.md]: on training set -> on the training set
on validation set -> on the validation set

File: image_classification/README.en.md [hunk 5]
```diff
@@ -395,106 +313,131 @@ The following are the components of `resnet_cifar10`:
...
 
-After training is done, the model from each pass is saved in `output/pass-%05d`. For example, the model of Pass 300 is saved in `output/pass-00299`. The script `classify.py` can be used to extract features and to classify an image. The default config file of this script is `models/vgg.py`.
+```python
+reader=paddle.reader.batch(
+    paddle.reader.shuffle(
+        paddle.dataset.cifar.train10(), buf_size=50000),
+        batch_size=128)
+```
 
+`feeding` is devoted to specify the correspondence between each yield record and `paddle.layer.data`. For instance,
```
Comment (helinwang) [inline image_classification/README.en.md]: is devoted to specify -> is devoted to specifying （devoted to something:)）
Or:
is used to specify

File: image_classification/README.en.md (hunk 3 above)
Comment (helinwang) [inline image_classification/README.en.md]: we uses a -> we use a

File: image_classification/README.en.md [hunk 6]
```diff
@@ -141,157 +141,87 @@ Figure 10. ResNet model for ImageNet
...
+        name="image", type=paddle.data_type.dense_vector(datadim))
 	```
 
 2. Define VGG main module
 
 	```python
-	net = vgg_bn_drop(data)
+	net = vgg_bn_drop(image)
 	```
         The input to VGG main module is from data layer. `vgg_bn_drop` defines a 16-layer VGG network, with each convolutional layer followed by BN and dropout layers. Here is the definition in detail:
```
Comment (helinwang) [inline image_classification/README.en.md]: from data layer -> from the data layer

File: image_classification/README.en.md [hunk 7]
```diff
@@ -141,157 +141,87 @@ Figure 10. ResNet model for ImageNet
...
```
Comment (helinwang) [inline image_classification/README.en.md]: 有一句在原文里不在diff里，也请改一下吧：
between input and output are different -> between input and output is different

File: image_classification/README.en.md [hunk 8]
```diff
@@ -141,157 +141,87 @@ Figure 10. ResNet model for ImageNet
...
-    learning_method=MomentumOptimizer(0.9),
-    regularization=L2Regularization(0.0005 * 128),)
+# PaddlePaddle init
+paddle.init(use_gpu=False, trainer_count=1)
 ```
 
-The learning rate adjustment policy can be defined with variables `learning_rate_decay_a`($a$), `learning_rate_decay_b`($b$) and `learning_rate_schedule`. In this example, discrete exponential method is used for adjusting learning rate. The formula is as follows,
-$$  lr = lr_{0} * a^ {\lfloor \frac{n}{ b}\rfloor} $$
-where $n$ is the number of processed samples, $lr_{0}$ is the learning_rate set in 'settings'.
+As alluded to in section `Overview Models`, here we provide the implementations of both VGG and ResNet models.
```
Comment (helinwang) [inline image_classification/README.en.md]:
```
`Overview Models` -> [Model Overview](#model-overview)
```

**Review by helinwang — CHANGES_REQUESTED** [review]

File: image_classification/README.en.md [hunk 9]
```diff
@@ -395,106 +313,131 @@ The following are the components of `resnet_cifar10`:
...
 
 ## Model Training
 
-We can train the model by running the script train.sh, which specifies config file, device type, number of threads, number of passes, path to the trained models, etc,
+### Define Parameters
 
-``` bash
-sh train.sh
-```
+First, we define the model parameters according to the previous model configuration `cost`.
```
Comment (helinwang) [inline image_classification/README.en.md]: define -> create

File: image_classification/README.en.md [hunk 10]
```diff
@@ -395,106 +313,131 @@ The following are the components of `resnet_cifar10`:
...
+# Create parameters
+parameters = paddle.parameters.create(cost)
 ```
 
-- `--config=$cfg` : specifies config file. The default is `models/vgg.py`.
-- `--use_gpu=true` : uses GPU for training. If use CPU，set it to be false.
-- `--trainer_count=1` : specifies the number of threads or GPUs.
-- `--log_period=100` : specifies the number of batches between two logs.
-- `--save_dir=$output` : specifies the path for saving trained models.
+### Construct Trainer
```
Comment (helinwang) [inline image_classification/README.en.md]: 因为后面有个
```
# Create trainer
```
这里感觉也写成### Create Trainer比较合适。

File: image_classification/README.en.md [hunk 11]
```diff
@@ -395,106 +313,131 @@ The following are the components of `resnet_cifar10`:
...
+ the first column of data generated by `cifar.train10()` corresponds to image layer's feature.
 
-### Prediction
+```python
+feeding={'image': 0,
+         'label': 1}
+```
 
-We can run the following script to predict the category of an image. The default device is GPU. If to use CPU, set `-c`.
+Callback function `event_handler` is used to track training and testing process that might be triggered once the action to which it is attached is executed.
```
Comment (helinwang) [inline image_classification/README.en.md]: 这句话写得语法有点太复杂了，我隐隐觉得可能不对，但又说不出来哪里。
要不要考虑简单处理：
Callback function event_handler will be called during training when a pre-defined event happens.

File: image_classification/README.en.md (hunk 1 above)
Comment (gangliao) [reply image_classification/README.en.md]: done

File: image_classification/README.en.md (hunk 2 above)
Comment (gangliao) [reply image_classification/README.en.md]: download

File: image_classification/README.en.md (hunk 8 above)
Comment (gangliao) [reply image_classification/README.en.md]: download

File: image_classification/README.en.md (hunk 3 above)
Comment (gangliao) [reply image_classification/README.en.md]: download

Comment (gangliao) [reply image_classification/README.en.md]: done

File: image_classification/README.en.md (hunk 6 above)
Comment (gangliao) [reply image_classification/README.en.md]: done

File: image_classification/README.en.md (hunk 7 above)
Comment (gangliao) [reply image_classification/README.en.md]: done

File: image_classification/README.en.md (hunk 5 above)
Comment (gangliao) [reply image_classification/README.en.md]: done

File: image_classification/README.en.md (hunk 4 above)
Comment (gangliao) [reply image_classification/README.en.md]: done

File: image_classification/README.en.md (hunk 9 above)
Comment (gangliao) [reply image_classification/README.en.md]: done

File: image_classification/README.en.md (hunk 11 above)
Comment (gangliao) [reply image_classification/README.en.md]: done

File: image_classification/README.en.md (hunk 10 above)
Comment (gangliao) [reply image_classification/README.en.md]: done

File: image_classification/README.en.md [hunk 12]
```diff
@@ -343,47 +268,38 @@ def conv_bn_layer(input,
...
-    short = shortcut(ipt, ch_in, ch_out, stride)
-    return addto_layer(input=[ipt, short], act=ReluActivation())
-
-def bottleneck(ipt, ch_out, stride):
-    ch_in = ipt.num_filter
-    tmp = conv_bn_layer(ipt, ch_out, 1, stride, 0)
-    tmp = conv_bn_layer(tmp, ch_out, 3, 1, 1)
-    tmp = conv_bn_layer(tmp, ch_out * 4, 1, 1, 0, LinearActivation())
+    tmp = conv_bn_layer(tmp, ch_out, 3, 1, 1, paddle.activation.Linear())
     short = shortcut(ipt, ch_in, ch_out, stride)
```
Comment (gangliao) [inline image_classification/README.en.md:295]: @qingqing01  请确认一下 到底是之前写错了 还是我写错了

Comment (gangliao) [reply image_classification/README.en.md:295]:
```
paddle.layer.addto(input=[tmp, short],
```

Comment (gangliao) [reply image_classification/README.en.md:295]: 确认过了 新修改的是对的

File: image_classification/index.en.html [hunk 13]
```diff
@@ -437,106 +353,132 @@
...
+    if isinstance(event, paddle.event.EndIteration):
+        if event.batch_id % 100 == 0:
+            print "\nPass %d, Batch %d, Cost %f, %s" % (
+                event.pass_id, event.batch_id, event.cost, event.metrics)
+        else:
+            sys.stdout.write('.')
+            sys.stdout.flush()
+    if isinstance(event, paddle.event.EndPass):
+        result = trainer.test(
+            reader=paddle.reader.batch(
```
Comment (qingqing01) [inline image_classification/index.en.html]: paddle.reader.batch -> paddle.batch

File: image_classification/index.en.html [hunk 14]
```diff
@@ -437,106 +353,132 @@
...
+                paddle.dataset.cifar.test10(), batch_size=128),
+            reader_dict={'image': 0,
```
Comment (qingqing01) [inline image_classification/index.en.html]: reader_dict -> feeding

Comment (qingqing01) [reply image_classification/index.en.html]: 这两处同时更改下中文吧~

File: image_classification/index.en.html (hunk 13 above)
Comment (gangliao) [reply image_classification/index.en.html]: done

File: image_classification/index.en.html (hunk 14 above)
Comment (gangliao) [reply image_classification/index.en.html]: done

**Review by helinwang — APPROVED** [review]
LGTM.
